### DISCUSSION TITLE
Running a node in TCP only

### DISCUSSION THREAD

**Discussion by SWvheerden** [tari-project/tari#6366] [Nodes / Mining / Wallet Support] [answered]
By default, the nodes will only in tor mode. But running a node in pure TCP is possible. 

To run in pure TCP do the following:

- Open up your config file and go to the section called: `[base_node.p2p]`
- Enable the option `public_addresses` and fill in your public internet IP and keep the port the same, example change `#public_addresses = ["/ip4/172.2.3.4/tcp/18189",]` to `public_addresses = ["/ip4/x.x.x.x/tcp/18189",]`  where `x.x.x.x` is you public internet IP. This step can be skipped. But it is important see, below. 
- scroll down to the section: `[base_node.p2p.transport]`
- Change `#type = "tor"` to `type = "tcp"`

Doing this will allow your node to connect to other TCP nodes. 

If you want your node to diable by other tcp nodes, you need to set the `public_address` section of the config above. If this option is not set, you can still dial other nodes, but you will not be open to other nodes dialing you and it can limit your connectivity. 

To completely set this up correctly, you need to enable port forwarding on your internet router. 
This step if different for each router type, but a check google search for how to do port forwarding on your router should give you a step-by-step guide. You need to port forward port `18189` to the IP of the PC running the base node.

**Comment by stringhandler** [comment, marked as answer]
See above
